File: companion/spss/pilot-b01/b01.py
import pandas as pd
import numpy as np
from scipy import stats

d = pd.read_csv("companion/spss/csv/b01.csv")
assert not d.isna().any().any()
d["school"] = pd.Categorical(d.school.map({1:"GP", 2:"MS"}))
d["sex"] = pd.Categorical(d.sex.map({1:"F", 2:"M"}))
d["studytime"] = pd.Categorical(
    d.studytime, categories=[1, 2, 3, 4], ordered=True)
d.info()
print(d.isna().sum())
print(d.school.value_counts()); print(d.sex.value_counts())
print(d[["age", "g1", "g2", "g3"]].agg(
    ["count", "mean", "std", "min", "max"]))

import sys, scipy
print("Python:", sys.version)
print("pandas:", pd.__version__)
print("numpy:", np.__version__)
print("scipy:", scipy.__version__)

File: companion/spss/pilot-b01/b01.R
d <- read.csv("companion/spss/csv/b01.csv")
stopifnot(!anyNA(d))
d$school <- factor(d$school, 1:2, c("GP", "MS"))
d$sex <- factor(d$sex, 1:2, c("F", "M"))
d$studytime <- ordered(d$studytime, levels=1:4)
str(d)
print(colSums(is.na(d)))
print(table(d$school)); print(table(d$sex))
x <- d[c("age", "g1", "g2", "g3")]
print(sapply(x, function(v) c(n=length(v),
  mean=mean(v), sd=sd(v), min=min(v), max=max(v))))